packages/node-runtime-worker-thread/src/worker-runtime.spec.ts

```diff
@@ -535,7 +535,22 @@ describe('worker', () => {
         await evaluate('print("Hi!")');
 
         expect(evalListener.onPrint).to.have.been.calledWith([
-          { printable: 'Hi!', rawValue: 'Hi!', type: null }
+          // TODO: should this be a ShellResult or a RuntimeEvaluationResult
+          { printable: 'Hi!', source: undefined, type: null }
```

Comment:
Shouldn’t there be a corresponding deserialization call on the receiving side?

Reply:
I thought so, but so far it seems to just work. Trying to see what happens in compass. Or do you mean turn RuntimeEvaluationResult into something else? I'm very confused.

Reply:
I assume that that might be because `printable` and `type` are preserved, but it is an API change – it means that an `onPrint` listener for consumers of node-runtime-worker-thread consumes a different API than an `onPrint` listener for other shell-api consumers.

I don’t know how bad this really is either, but it’s definitely not what the API is expected to be.

Reply:
Yeah, as Anna said it will sorta work because the ShellResult type shape is preserved and browser repl will try it's best to show stuff on the screen, but you'd want to use `deserializeEvaluationResult` in child-process-evaluation-listener:

https://github.com/mongodb-js/mongosh/blob/857a89679cf62b9dc29d5400c65a31f2b6f5ef41/packages/node-runtime-worker-thread/src/child-process-evaluation-listener.ts#L17-L19

Reply:
I've added that in now. Problem is the way the tests are written that's never called because the tests provide their own onPrint. At least the ones I've been working on. I'll see how hard that is to change.